contributor-docs/adrs/adr-003-prop-norms.md

```diff
@@ -0,0 +1,70 @@
+# ADR 003: Prop norms in Primer React Components
+
+## Status
+
+Proposed
+
+## Context
+
+Today we have:
+
+- Implicit conventions not documented anywhere but consistently reflected in our code (e.g., the type of the `sx` prop)
+- Explicit plans to change some of those (e.g., the deprecation of styled-system props)
+- Inconsistencies in our implementation (e.g., when components accept a `ref` prop)
+
+This ADR aims to unify conversations about prop APIs, codify our decisions, and sequence the work to get there.
+
+## Decision
+
+### 🟢 `sx`
+
+All components that ultimately render to the DOM should accept an `sx` prop.
+
+The `sx` prop (of type `SystemStyleObject`) should generally set styles for the root HTML element rendered by the component. An exception would be components like `Dialog`, whose outermost HTML element is a backdrop. In that case, it would be appropriate for `sx` styles to apply to child of the backdrop that is more likely to need styling overrides.
+
+It may make sense to have an additional prop to style a descendant element (e.g., `Dialog` could accept a `backdropSx` prop).
```

Comment:
`backdropSx` feels wrong to me for some reason 🤔  I'm unsure if we'd need to allow style overrides for the backdrop.

Reply:
For `Dialog`, should `ref` be passed to the dialog or the backdrop?

Reply:
> backdropSx feels wrong to me for some reason 🤔 I'm unsure if we'd need to allow style overrides for the backdrop.

Fair. Can you think of any place where a second `sx` prop would feel right?

> For Dialog, should ref be passed to the dialog or the backdrop?

Good point. I'd expect it to go to the dialog. See my above note on weakening the language re: who gets the `ref`.